### PR TITLE
feat: 💥 UI WG PRs

### LINKED ISSUE #33: Menu/Search isn't working.
It redirects to /tz/ and doesn't show content.

### PR DESCRIPTION
**Pull request by melMass**

### CODE REVIEW

File: src/components/item-info/index.js [hunk 1]
```diff
@@ -126,65 +126,32 @@ export const ItemInfo = ({
...
-            </div>
-            <Button onClick={() => handleCollect()}>
+            <p style={{ paddingBottom: '7.5px' }}>OBJKT#{id}</p>
+            <Button onClick={() => handleCollect()} full>
               <Purchase>{message}</Purchase>
             </Button>
           </div>
         )}
-        <div className={styles.spread}>
+        {/* <div className={styles.spread}>
```
Comment (xat) [inline src/components/item-info/index.js:148]: maybe remove the block, if it's not used?

**Comment by melMass** [comment]
A few things from the deployment that I can see:

- The menu needs to be update (missing collaborate and friends), the components/pages are there just not referenced in the menu

- The price indicator, should it look like that?
image
image

- For the menu when the text is longer maybe the font-weight should be bolder? @sf-pixel 
image

**Comment by sf-pixel** [comment]
iirc the price indicator is as the figma indicates, the menu needs tweaking though.

**Comment by melMass** [comment]
> iirc the price indicator is as the figma indicates

I'll ask Dens on the discord about the design stuff! 
If you can test it a bit and check if everything seems there, I can handle the edits!

File: src/context/HicetnuncContext.js [hunk 2]
```diff
@@ -288,6 +288,75 @@ class HicetnuncContextProviderClass extends Component {
           '--shadow-color',
           light ? 'rgba(0,0,0,0.2)' : 'rgba(255,255,255,0.3)'
         )
+        root.style.setProperty(
```
Comment (xat) [inline src/context/HicetnuncContext.js:291]: i don't think it needs to be changed as part of this pr, but imo we should (at some point) find a better way to switch between dark and light mode that doesn't involve a dozen if/else statements.

File: src/pages/objkt-display/tabs/history.js [hunk 3]
```diff
@@ -19,30 +20,200 @@ export const History = (token_info) => {
 
     let history = [...trades, ...swaps].sort((a, b) => Date.parse(a.timestamp) - Date.parse(b.timestamp)).reverse()
 
+    const getTimeAgo = (props) => {
```
Comment (xat) [inline src/pages/objkt-display/tabs/history.js]: could be moved into `utils/time.js`

File: src/context/HicetnuncContext.js (hunk 2 above)
Comment (melMass) [reply src/context/HicetnuncContext.js:291]: Yes like a `.dark` class, btw there is a few unused stuff again from this PR 🚑  😅 

As I get notifications feel free to push to it too, I'll know if I need to pull

File: src/pages/objkt-display/tabs/history.js [hunk 4]
```diff
@@ -19,30 +20,200 @@ export const History = (token_info) => {
...
+                                                    {parseFloat(e.swap.price / 1e6)} tez
+                                                </div>
+                                            </div>
+                                        </div>
+                                    )
+                                } else {
+                                    return (
+                                        <div className={`${styles.history}`} key={`s-${e.ophash}`}>
+                                            <div className={styles.history__event__container}>
+                                                <svg width="16" height="16" viewBox="0 0 19 19" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (xat) [inline src/pages/objkt-display/tabs/history.js]: could be moved into a separate component

File: src/pages/objkt-display/tabs/history.js [hunk 5]
```diff
@@ -19,30 +20,200 @@ export const History = (token_info) => {
...
+                                            </div>
+                                        </div>
+                                    )
+                                }
+                            })
+                        }
+
+                        <div className={styles.history} key="mint-op">
+                            <div className={styles.history__event__container}>
+                                <svg style={{ fill: 'none' }} width="16" height="16" viewBox="0 0 16 15" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (xat) [inline src/pages/objkt-display/tabs/history.js]: could be moved into a separate component

File: package.json [hunk 6]
```diff
@@ -13,6 +13,7 @@
     "axios": "^0.21.1",
     "base-x": "^3.0.8",
     "buffer": "^6.0.3",
+    "chokidar": "^3.5.2",
```
Comment (xat) [inline package.json]: is this dependency used somewhere?

File: src/components/header/index.js [hunk 7]
```diff
@@ -24,7 +24,7 @@ const wallet = new BeaconWallet({
 export const Header = () => {
   const history = useHistory()
   const context = useContext(HicetnuncContext)
-
+  const style = {fontSize : '20px'}
```
Comment (xat) [inline src/components/header/index.js]: does this need to be a separate variable? maybe just set the style inline?

File: src/components/item-info/index.js (hunk 1 above)
Comment (xat) [inline src/components/item-info/index.js]: if the code is not needed anymore, better to just remove.

**Comment by melMass** [comment]
@xat I think we can make all those edits in this PR indeed (component separation etc...) 
Let's just wait for pixel and dens to confirm it's all there first but we will do the cleaning in this branch

**Comment by xat** [comment]
yes, makes sense.

File: package.json (hunk 6 above)
Comment (melMass) [reply package.json]: It's a sub dependency of `react-scripts` my guess is that it was there to enforce a specific version.

**Comment by LeonNicholls** [comment]
All the FAQ links are wrong: https://teia-main-pr-17.onrender.com/faq

**Comment by melMass** [comment]
> All the FAQ links are wrong: https://teia-main-pr-17.onrender.com/faq

That's already fixed in teia-community/teia-ui#21

**Comment by LeonNicholls** [comment]
Menu/Search isn't working.

**Comment by jagracar** [comment]
What is the transfer tab, after the burn tab, supposed to do? It says i don't have editions to transfer, but it have many unswapped editions. The burn tab seems to work

**Comment by jagracar** [comment]
swap, burn and transfer seem to have a different indentation than info, listings and history. 
I think in the other cases the text is inside a container with margin: 0px 1em; while that container is missing in swap, burn and transfer.

**Comment by xat** [comment]
> Menu/Search isn't working.

hmm, true. it currently just redirects to `http://localhost:3000/tz/`

**Comment by xat** [comment]
i made a fresh rebase of this pr on top of main, so it now contains the latest changes that were merged into main in the meantime (including the updated faq links).

**Comment by LeonNicholls** [comment]
Collabs page displays a loading indicator forever when you don't have collabs.

**Comment by melMass** [comment]
> collabs

Let's focus on what this PR brings, I think this issue exists in teia already.

Also transfer is part of the collab code somehow. That will require it's dedicated PR, can you file an issue?

**Comment by LeonNicholls** [comment]
So this UI PR doesn't affect the collabs? Are there other pages not affected?
Filed issue:
teia-community/teia#2

**Comment by LeonNicholls** [comment]
Issue for menu/search: https://github.com/teia-community/teia-ui/issues/33

**Comment by melMass** [comment]
> What is the transfer tab, after the burn tab, supposed to do? It says i don't have editions to transfer, but it have many unswapped editions. The burn tab seems to work

It is part of the collab code... It needs to be cleaned up and added to `objkt-display`.

I fixed the `Search` route and the mismatching margins

**Comment by melMass** [comment]
I fixed what was talked about in the comments here.

**Comment by xat** [comment]
i agree with @melMass: the ui has a lot of bugs, but they mostly weren't introduced by this pr.

**Review by xat — COMMENTED** [review]
i tested it a bit more and think that it is good to merge. we can tackle any remaining issues in follow-ups.

**Review by xat — APPROVED** [review]

**Comment by melMass** [comment]
> i tested it a bit more and think that it is good to merge. we can tackle any remaining issues in follow-ups.

Sounds good to me!

**Pull request merged**